scRNA-seq/setup/tabula-muris/combine_alevin.Rmd

```diff
@@ -0,0 +1,255 @@
+
+```{r}
+qc_df <- tm_sces %>%
+  purrr::map_df(~ as.data.frame(colData(.x)))
+```
+
+#### Scatter plot of each sample
+```{r}
+ggplot(qc_df, aes(x = total, y = detected, color = subsets_mito_percent)) + 
+  geom_point(alpha = 0.2) +
+  scale_color_viridis_c() +
+  facet_wrap(~ sample_id) 
+```
+
+#### Density plots of mitochondrial percentages
+```{r}
+ggplot(qc_df, aes(x = subsets_mito_percent, color = tissue, fill = tissue)) + 
+  geom_density() + 
+  facet_wrap(~ sample_id, scales = "free_y") 
+```
+
+### Filter cells based on QC measures
+
+This may not be ideal, but for now lets set the following cutoffs:
+
+- Mitochondrial percent < 25%
+- detected genes > 1000
+
+
+```{r}
+filter_sce <- function(sce, detect_threshold, mito_threshold) {
+  qc_pass <- ((sce$detected > detect_threshold) 
+              & (sce$subsets_mito_percent < mito_threshold))
+  sce_filtered <- sce[, qc_pass]
+  return(sce_filtered)
+}
+
+tm_filtered <- tm_sces %>%
+  purrr::map(filter_sce, detect_threshold = 1000, mito_threshold = 25) 
+```
+
+
+### Combine SingleCellExperiments
+
+This is a naive combination, simply concatinating the component samples into one SCE class/expression matrix.
+If we had batch info, we might want to apply that to do better.
+
+```{r}
+tm_combined <- do.call(cbind, tm_filtered)
+
+# save some memory
+rm(tm_filtered, tm_sces)
+```
+
+### Feature (gene) QC
+
+Doing the Feature QC at this point is required because low abundance/uncommon genes can result in negative size factor estimates in the normalization steps, which is undesirable, to say the least.
+
+
+```{r}
+tm_combined <- addPerFeatureQC(tm_combined)
+```
+
+Feature filtering criteria: 
+- Genes found in at least 20 cells
+- mean count of at least 0.01 
+
+```{r}
+detected <- rowData(tm_combined)$detected > 20
+expressed <- rowData(tm_combined)$mean > 0.01
+
+# filter the genes (rows) this time
+tm_combined <- tm_combined[detected & expressed, ]
+```
+
+
+## Normalization
+
+We are normalizing with `scran::computeSumFactors()` which implements the strategy of [Lun 
+*et al.* (2016)](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7).
+
+```{r normalize}
+# Cluster similar cells
+qclust <- scran::quickCluster(tm_combined)
+
+# Step 2) Compute sum factors for each cell cluster grouping.  
+tm_combined <- scran::computeSumFactors(tm_combined, clusters = qclust)
+
+# Step 3) Normalize using these pooled sum factors and log transform. 
+tm_combined <- scater::logNormCounts(tm_combined)
+```
+
+
+## Variance filtering
+
+Since we are using this as an example data set, we are going to shrink it down past what we might normally do.
+We will select the 1000 genes with the highest biologial variance across samples, after modeling the technical variance using `scran::modelGeneVar()`
+
+```{r}
+gene_var <- scran::modelGeneVar(tm_combined)
+highvar_genes <- scran::getTopHVGs(gene_var, n = 1000)
+
+# filter
+tm_combined_filt <- tm_combined[highvar_genes, ]
+```
```

Comment:
I can't tell if I would "normally" do this before or after combining tissues but the answer to that question is probably it depends.